packages/edit-site/src/components/keyboard-shortcuts/global.js

```diff
@@ -28,7 +32,7 @@ function KeyboardShortcutsGlobal() {
 			isSavingEntityRecord( record.kind, record.name, record.key )
 		);
 		const _hasNonPostEntityChanges = hasNonPostEntityChanges();
-		const isViewMode = getCanvasMode() === 'view';
+		const isViewMode = canvas === 'view';
```

Comment:
This is inside an event handler, you could read the `canvas` value directly and non-reactively with `history.getLocationWithParams().params.canvas`.